rosette/src/BaseSupp.cc

```diff
@@ -742,24 +744,29 @@ DEF("regexpCompare", regexpCompare, 2, 2) {
     CHECK(0, RBLstring, re);
     CHECK(1, RBLstring, str);
 
-    char *re_s, *str_s;
-    regexp* r;
     Ob* rv;
 
-    re_s = (char*)&re->byte(0);
-    str_s = (char*)&str->byte(0);
+    char* re_s = (char*)&re->byte(0);
+    char* str_s = (char*)&str->byte(0);
+
+    try {
```

Comment:
I strongly disfavor C++ exceptions. Is there a compelling reason you want a `try` here?

Reply:
Some of the documentation that I read hinted that the std::regex constructor might throw exceptions on invalid expressions. If you don't put it in a try catch block then user input (i.e. the expression) could cause the VM to bail out. The try catch blocks catch this and return the error to the user's code.